reactive/apache_bigtop_base.py

```diff
@@ -19,3 +32,20 @@ def fetch_bigtop():
         hookenv.status_set('waiting',
                            'Unable to fetch Bigtop: checksum error'
                            ' (will retry)')
+
+
+@when_any('java.ready', 'hadoop-plugin.java.ready')
+def set_java_home():
```

Comment:
Similar exception here: JavaProvides param is missing

unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.juju-log server.go:268 java:6: Invoking reactive handler: reactive/apache_bigtop_base.py:38:set_java_home
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40 Traceback (most recent call last):
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40   File "/var/lib/juju/agents/unit-resourcemgr-0/charm/hooks/java-relation-changed", line 19, in <module>
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40     main()
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/__init__.py", line 69, in main
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40     bus.dispatch()
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 418, in dispatch
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40     _invoke(other_handlers)
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 401, in _invoke
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40     handler.invoke()
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 277, in invoke
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40     self._action(*args)
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 INFO unit.resourcemgr/0.java-relation-changed logger.go:40 TypeError: set_java_home() takes 0 positional arguments but 1 was given
unit-resourcemgr-0[2549]: 2016-05-17 11:24:34 ERROR juju.worker.uniter.operation runhook.go:107 hook "java-relation-changed" failed: exit status 1
(E